File: 0125-valid-palindrome/0125-valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    let fs = s.split(/[^a-zA-Z0-9]+/).join('').toLowerCase()
    let i = 0, j = fs.length -1
    while (i<j && j>i){
        if (fs[i] !== fs[j]) {return false}
        else{
            i++
            j--
        }
    }
    return true
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.toLowerCase();
        int i = 0;
        int j = s.length() -1;

        while (i < j){
            char ci = s.charAt(i);
            char cj = s.charAt(j);

            if (!Character.isLetterOrDigit(ci)){
                i++;
            } else if (!Character.isLetterOrDigit(cj)){
                j--;
            } else {
                if (ci != cj){
                    return false;
                } else {
                    i++;
                    j--;
                }
            }
        }
        return true;
    }
}